CONTRIBUTING.md

```diff
@@ -21,33 +21,54 @@ All submissions, including submissions by project members, require review.
 We use GitHub pull requests for this purpose. Consult GitHub Help for more
 information on using pull requests.
 
+Proper code review should include proper unit tests with them. Please see
+[Testing Conventions](##Testing Conventions) for our conventions on how to write proper unit tests.
+
+TODO(zhitaoli): Configure a CI/CD on Github so that each PR submission is tested
+before merged.
+
 ## Community Guidelines
 This project follows[Google's Open Source Community Guidelines](
 https://opensource.google.com/conduct/).
 
 # Contributing Guidelines
+At this point, TFX only supports Python 3 on Linux and MacOS. Please use one of
+these operation system for development and testing.
 
-## Running Unit Tests
+If Python 3.5 is used, our usage of type hints requires at least 3.5.3.
 
-We recommend using virtualenv and Python 3.6 and above for development. You can
-invoke all Python unit tests:
+## Testing Conventions
+All python unit tests in this repo is based on Tensorflow's [tf.test.TestCase](
+https://www.tensorflow.org/api_docs/python/tf/test/TestCase), which is a subclass
+of [py-absl TestCase](https://github.com/abseil/abseil-py/blob/06edd9c20592cec39178b94240b5e86f32e19768/absl/testing/absltest.py#L523).
 
-```
-python setup.py test
-```
-
-By default, only unit tests are executed. Tests marked as `end_to_end` will be
-skipped.
+We have several types of tests in this repo:
+*   Unit tests for source code;
+*   End to end tests (filename ends with `_e2e_test.py`): some of this also runs
+    with external environments.
 
 ## Testing local change
 To test local change, you will need to install code into virtualenv in editable
 mode:
 
-```
+```shell
 pushd <your_source_dir>
-pip install -e .
+pip install -e .[all]   # the [all] suffix includes additional packages for test
```

Comment:
Because [all] installs python-snappy as well, it [requires snappy C library as well](https://github.com/andrix/python-snappy#dependencies). I thought that it would be more convenient if we don't have such a dependency.

So I'd like to propose a new "extra" configuration called "test". 
For example: https://github.com/tensorflow/tfx/blob/b0e93a3817bdba23e02d69c3695f3f108af82913/setup.py#L127

Reply:
Per #1679, we can rely on `'all'`.